src/include/distributed/metadata_cache.h

```diff
@@ -38,11 +38,12 @@ typedef struct
 
 	/* pg_dist_shard metadata (variable-length ShardInterval array) for this table */
 	int shardIntervalArrayLength;
-	ShardInterval *shardIntervalArray;
+	ShardInterval **sortedShardIntervalArray;
```

Comment:
I have a question here. 

I don't see any drawbacks of keeping an array of shard interval pointers instead of array of shard intervals. But, do you think are there any drawbacks that I might have missed?